src/routes/pdf-fetcher/containers/PdfFetcherPage.js

```diff
@@ -31,14 +31,18 @@ class PdfFetcherPage extends React.Component {
   }
 
   componentWillMount () {
-    this.fetchUrl(CategoryModel.getCategoryByPath(this.props.categories, this.props.path))
+    const category = this.props.categories.getCategoryByUrl(this.props.fetchUrl)
+    if (category) {
+      this.fetchUrl(category)
+    } else {
+      // todo display an error
+    }
   }
 
-  addCategoryIdsRecursively (categoryIds, currentCategory) {
-    currentCategory.children.forEach((id) => {
-      const child = CategoryModel.getCategoryById(this.props.categories, id)
+  addCategoryIdsRecursively (categoryIds, children) {
+    children.forEach((child) => {
       categoryIds.push(child.id)
-      this.addCategoryIdsRecursively(categoryIds, child)
+      this.addCategoryIdsRecursively(categoryIds, this.props.categories.getChildren(child.url))
     })
   }
```

Comment:
With "un-flatten-tree" this could be refactored to
```javascript
getCategoryIds (category){
  const ids = flatten(
       [category],
       _category => this.props.categories.getChildren(_category))
     .map(_category => _category.id)
  if(category.id === 0) {
    ids.shift()
  }
  return ids
}
```

Reply:
jop, we should discuss this on slack